h5py/h5d.pyx

```diff
@@ -497,3 +503,92 @@ cdef class DatasetID(ObjectID):
+        def get_num_chunks(self, SpaceID space=None):
+            """ (SpaceID space=None) => INT num_chunks
+
+            Retrieve the number of chunks that have nonempty intersection with a
+            specified dataspace. Currently, this function only gets the number
+            of all written chunks, regardless of the dataspace.
+
+            Feature requires: 1.10.5 HDF5
+            """
+            cdef hsize_t num_chunks
+
+            if space is None:
+                space = self.get_space()
+            H5Dget_num_chunks(self.id, space.id, &num_chunks)
+            return num_chunks
+
+        @with_phil
+        def get_chunk_info(self, hsize_t index, SpaceID space=None):
+            """ (hsize_t index, SpaceID space=None) => StoreInfo
+
+            Retrieve storage information about a chunk specified by its index.
+
+            Feature requires: 1.10.5 HDF5
+            """
+            cdef haddr_t byte_offset
+            cdef hsize_t size
+            cdef hsize_t *chunk_offset
+            cdef unsigned filter_mask
+            cdef hid_t space_id = 0
+            cdef int rank
+
+            if space is None:
+                space_id = H5Dget_space(self.id)
+            else:
+                space_id = space.id
+
+            rank = H5Sget_simple_extent_ndims(space_id)
+            chunk_offset = <hsize_t*>emalloc(sizeof(hsize_t) * rank)
+            H5Dget_chunk_info(self.id, space_id, index, chunk_offset,
+                              &filter_mask, &byte_offset, &size)
+            cdef tuple cot = convert_dims(chunk_offset, <hsize_t>rank)
+            efree(chunk_offset)
+
+            if space is None:
+                H5Sclose(space_id)
+
+            return StoreInfo(index,
+                             cot if byte_offset != HADDR_UNDEF else None,
+                             filter_mask,
+                             byte_offset if byte_offset != HADDR_UNDEF else None,
+                             size)
+
+
+        @with_phil
+        def get_chunk_info_by_coord(self, tuple chunk_offset not None):
+            """ (TUPLE chunk_offset) => StoreInfo
+
+            Retrieve information about a chunk specified by the array
+            address of one of its dataset elements.
+
+            Feature requires: 1.10.5 HDF5
+            """
+            cdef haddr_t byte_offset
+            cdef hsize_t size
+            cdef unsigned filter_mask
+            cdef hid_t space_id = 0
+            cdef int rank
+            cdef hsize_t *co = NULL
+
+            space_id = H5Dget_space(self.id)
+            rank = H5Sget_simple_extent_ndims(space_id)
+            H5Sclose(space_id)
+            co = <hsize_t*>emalloc(sizeof(hsize_t) * rank)
+            convert_tuple(chunk_offset, co, rank)
+            H5Dget_chunk_info_by_coord(self.id, co,
+                                       &filter_mask, &byte_offset,
+                                       &size)
+            cdef tuple cot = convert_dims(co, <hsize_t>rank)
```

Comment:
The offset is only [documented](https://confluence.hdfgroup.org/display/HDF5/H5D_GET_CHUNK_INFO_BY_COORD) as an input parameter, and looking at the HDF5 code, I think that's correct. So converting the array back to a tuple is redundant - you can just use `chunk_offset` as passed in.

Reply:
Done.